### ISSUE TITLE
can not create file "main.relectable.dart" in flutter 2.0

### ISSUE THREAD

**Issue by wenboLee** [google/reflectable.dart#246]
i think the problem is env.

<details>
  <summary>flutter pub run build_runner build -v</summary>

  ``` log
flutter pub run build_runner build -v
[  +88 ms] executing: [/Users/supermonkey/Soft/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +67 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] b1395592de68cc8ac4522094ae59956dd21a91db
[   +1 ms] executing: [/Users/supermonkey/Soft/flutter/] git tag --points-at b1395592de68cc8ac4522094ae59956dd21a91db
[  +32 ms] Exit code 0 from: git tag --points-at b1395592de68cc8ac4522094ae59956dd21a91db
[        ] 2.0.4
[  +66 ms] executing: [/Users/supermonkey/Soft/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +15 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/Users/supermonkey/Soft/flutter/] git ls-remote --get-url origin
[  +15 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +62 ms] executing: [/Users/supermonkey/Soft/flutter/] git rev-parse --abbrev-ref HEAD
[  +16 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[   +6 ms] executing: sw_vers -productName
[  +17 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[  +16 ms] Exit code 0 from: sw_vers -productVersion
[        ] 11.2.3
[        ] executing: sw_vers -buildVersion
[  +17 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 20D91
[   +7 ms] executing: sysctl hw.optional.arm64
[   +5 ms] Exit code 1 from: sysctl hw.optional.arm64
[        ] sysctl: unknown oid 'hw.optional.arm64'
[  +88 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +54 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +42 ms] Using /Users/supermonkey/Soft/flutter/.pub-cache for the pub cache.
[        ] executing: /Users/supermonkey/Soft/flutter/bin/cache/dart-sdk/bin/pub run build_runner build -v
[INFO] Entrypoint:Generating build script...
[INFO] Entrypoint:Generating build script completed, took 680ms

[INFO] Bootstrap:Creating build script snapshot......
[FINE] Bootstrap:stdout: Info: Compiling without sound null safety
[INFO] Bootstrap:Creating build script snapshot... completed, took 14.4s

[INFO] Bootstrap:There was output on stdout while compiling the build script snapshot, run with `--verbose` to see it (you will need to run a `clean` first to re-snapshot).

[INFO] BuildDefinition:Initializing inputs
[INFO] BuildDefinition:Building new asset graph...
[INFO] BuildDefinition:Building new asset graph completed, took 1.0s

[INFO] BuildDefinition:Checking for unexpected pre-existing outputs....
[INFO] BuildDefinition:Checking for unexpected pre-existing outputs. completed, took 1ms

[INFO] Build:Running build...
[INFO] Heartbeat:1.4s elapsed, 0/16 actions completed.
[INFO] Heartbeat:2.4s elapsed, 1/17 actions completed.
[INFO] Heartbeat:3.5s elapsed, 2/18 actions completed.
[INFO] Heartbeat:4.5s elapsed, 2/18 actions completed.
[INFO] Heartbeat:5.6s elapsed, 3/19 actions completed.
[INFO] Heartbeat:6.7s elapsed, 3/19 actions completed.
[INFO] Heartbeat:7.7s elapsed, 3/19 actions completed.
[INFO] Heartbeat:8.7s elapsed, 3/19 actions completed.
[INFO] Heartbeat:9.8s elapsed, 3/19 actions completed.
[INFO] Heartbeat:16.3s elapsed, 4/20 actions completed.
[INFO] Heartbeat:17.4s elapsed, 5/21 actions completed.
[INFO] Heartbeat:20.3s elapsed, 8/24 actions completed.
[INFO] Heartbeat:25.0s elapsed, 14/30 actions completed.
[INFO] Heartbeat:30.9s elapsed, 17/33 actions completed.
[INFO] Heartbeat:31.9s elapsed, 37/53 actions completed.
[INFO] Heartbeat:32.9s elapsed, 52/68 actions completed.
[INFO] Heartbeat:34.0s elapsed, 66/81 actions completed.
[INFO] Heartbeat:35.0s elapsed, 80/95 actions completed.
[INFO] Heartbeat:36.0s elapsed, 94/110 actions completed.
[INFO] Heartbeat:37.2s elapsed, 109/125 actions completed.
[INFO] Heartbeat:38.2s elapsed, 127/143 actions completed.
[INFO] Heartbeat:39.2s elapsed, 139/154 actions completed.
[INFO] Heartbeat:40.2s elapsed, 153/169 actions completed.
[INFO] Heartbeat:41.3s elapsed, 172/188 actions completed.
[INFO] Heartbeat:42.4s elapsed, 187/203 actions completed.
[INFO] Heartbeat:43.4s elapsed, 202/217 actions completed.
[INFO] Heartbeat:44.4s elapsed, 220/236 actions completed.
[INFO] Heartbeat:45.4s elapsed, 233/249 actions completed.
[SEVERE] mockito:mockBuilder on lib/app/res/super_text_styles.dart:

Null check operator used on a null value
#0      LinkedElementFactory.isLibraryUri (package:analyzer/src/summary2/linked_element_factory.dart:418:48)
#1      LibraryContext.isLibraryUri (package:analyzer/src/dart/analysis/library_context.dart:96:27)
#2      LibraryAnalyzer._isLibrarySource (package:analyzer/src/dart/analysis/library_analyzer.dart:521:25)
#3      LibraryAnalyzer._resolveDirectives (package:analyzer/src/dart/analysis/library_analyzer.dart:562:36)
#4      LibraryAnalyzer.analyzeSync (package:analyzer/src/dart/analysis/library_analyzer.dart:127:5)
#5      LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:104:12)
#6      AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1315:63)
#7      PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#8      AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1286:20)
#9      AnalysisDriver._computeErrors (package:analyzer/src/dart/analysis/driver.dart:1369:26)
#10     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:987:20)
#11     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2006:24)
<asynchronous suspension>

package:analyzer/src/dart/analysis/driver.dart 1344:9             AnalysisDriver._computeAnalysisResult.<fn>
package:analyzer/src/dart/analysis/performance_logger.dart 32:15  PerformanceLog.run
package:analyzer/src/dart/analysis/driver.dart 1286:20            AnalysisDriver._computeAnalysisResult
package:analyzer/src/dart/analysis/driver.dart 1369:26            AnalysisDriver._computeErrors
package:analyzer/src/dart/analysis/driver.dart 987:20             AnalysisDriver.performWork
package:analyzer/src/dart/analysis/driver.dart 2006:24            AnalysisDriverScheduler._run

[INFO] Heartbeat:47.3s elapsed, 246/262 actions completed.
[INFO] Heartbeat:48.4s elapsed, 246/262 actions completed.
[INFO] Heartbeat:49.5s elapsed, 246/262 actions completed.
[INFO] Heartbeat:50.5s elapsed, 246/262 actions completed.
[INFO] Heartbeat:51.6s elapsed, 246/262 actions completed.
[INFO] Heartbeat:52.6s elapsed, 246/262 actions completed.
[INFO] Heartbeat:53.6s elapsed, 246/262 actions completed.
[INFO] Heartbeat:54.6s elapsed, 246/262 actions completed.
[INFO] Heartbeat:55.7s elapsed, 246/262 actions completed.
[INFO] Heartbeat:56.7s elapsed, 246/262 actions completed.
[INFO] Heartbeat:57.8s elapsed, 246/262 actions completed.
[INFO] Heartbeat:58.9s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 0s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 1s elapsed, 246/262 actions completed.
[WARNING] Heartbeat:
No actions completed for 15.1s, waiting on:
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/wx_binding_page.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/header_container.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/bottom_container.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/mid_container.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/wx_binding_btn.dart
  .. and 11 more

[INFO] Heartbeat:1m 2s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 3s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 4s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 5s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 6s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 7s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 8s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 9s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 10s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 11s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 13s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 14s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 15s elapsed, 246/262 actions completed.
[INFO] Heartbeat:1m 16s elapsed, 246/262 actions completed.
[WARNING] Heartbeat:
No actions completed for 15.1s, waiting on:
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/wx_binding_page.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/header_container.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/bottom_container.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/mid_container.dart
  - mockito:mockBuilder on lib/app/pages/common/wx_binding/widgets/wx_binding_btn.dart
  .. and 11 more

[INFO] Heartbeat:1m 17s elapsed, 246/262 actions completed.

Process finished with exit code 0.
```
  
</details>

<details>
  <summary>flutter doctor -v</summary>

``` bash
-> % flutter doctor -v
[✓] Flutter (Channel stable, 2.0.4, on macOS 11.2.3 20D91 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.0.4 at /Users/supermonkey/Soft/flutter
    • Framework revision b1395592de (11 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/supermonkey/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.55.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.114

• No issues found!
```
</details>

**Comment by eernstg** [comment]
Hi @wenboLee, what makes you think this is a reflectable issue? As far as I can see the failure happens in the mockito builder (so it should be reported at [mockito](https://github.com/dart-lang/mockito/issues)):

```
[SEVERE] mockito:mockBuilder on lib/app/res/super_text_styles.dart:

Null check operator used on a null value
#0      LinkedElementFactory.isLibraryUri (package:analyzer/src/summary2/linked_element_factory.dart:418:48)
#1      LibraryContext.isLibraryUri (package:analyzer/src/dart/analysis/library_context.dart:96:27)
#2      LibraryAnalyzer._isLibrarySource (package:analyzer/src/dart/analysis/library_analyzer.dart:521:25)
#3      LibraryAnalyzer._resolveDirectives (package:analyzer/src/dart/analysis/library_analyzer.dart:562:36)
#4      LibraryAnalyzer.analyzeSync (package:analyzer/src/dart/analysis/library_analyzer.dart:127:5)
#5      LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:104:12)
#6      AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1315:63)
#7      PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#8      AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1286:20)
#9      AnalysisDriver._computeErrors (package:analyzer/src/dart/analysis/driver.dart:1369:26)
#10     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:987:20)
#11     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2006:24)
<asynchronous suspension>
```

I'll close this issue now. Please create a new issue if additional evidence pointing at reflectable comes up.

**Comment by wenboLee** [comment]
ohhhhhhh, thank you very much.😂
